js/ui/status/automaticUpdates.js

```diff
@@ -187,9 +220,78 @@ var Indicator = new Lang.Class({
         if (currentActiveConnection)
-            this._settingChangedSignalId = currentActiveConnection.connect('changed', this._sync.bind(this));
+            this._settingChangedSignalId = currentActiveConnection.connect('changed', this._updateStatus.bind(this));
+    },
+
+    _updateNotification: function() {
+        if (this._state == AutomaticUpdatesState.UNKNOWN)
+            return;
+
+        let isNewConnection = this._isNewConnection();
+        let newState = this._getState();
+
+        let wasDisconnected = this._state == AutomaticUpdatesState.DISCONNECTED;
+        let wasActive = this._state >= AutomaticUpdatesState.IDLE;
+        let isActive = newState >= AutomaticUpdatesState.IDLE;
+
+        // The criteria to notify about the Automatic Updates setting is:
+        //   1. If the user was disconnected and connects to a new network; or
+        //   2. If the user was connect and connects to a network with different status;
+        if ((wasDisconnected && !isNewConnection) || (!wasDisconnected && isActive == wasActive))
+            return;
+
+        if (this._notification)
+            this._notification.destroy();
+
+        if (newState == AutomaticUpdatesState.DISCONNECTED)
+            return;
+
+        let source = new MessageTray.SystemNotificationSource();
+        Main.messageTray.add(source);
+
+        // Figure out the title, subtitle and icon
+        let title, subtitle, iconFile;
+
+        if (isActive) {
+            title = _("Automatic updates on");
+            subtitle = _("You have Unlimited Data so automatic updates have been turned on for this connection.");
+            iconFile = automaticUpdatesStateToString(AutomaticUpdatesState.IDLE);
+        } else {
+            title = _("Automatic updates are turned off to save your data");
+            subtitle = _("You will need to choose which Endless updates to apply when on this connection.");
+            iconFile = automaticUpdatesStateToString(AutomaticUpdatesState.DISABLED);
+        }
+
+        let gicon = new Gio.FileIcon({ file: Gio.File.new_for_uri(iconFile) });
+
+        // Create the notification
+        this._notification = new MessageTray.Notification(source, title, subtitle, { gicon: gicon });
+        this._notification.setUrgency(MessageTray.Urgency.NORMAL);
+        this._notification.setTransient(false);
+
+        this._notification.addAction(_("Change Settings…"), function() {
+            let app = Shell.AppSystem.get_default().lookup_app('gnome-updates-panel.desktop');
+            Main.overview.hide();
+            app.activate();
+        });
+
+        source.notify(this._notification);
+
+        this._notification.connect('destroy', function() {
+            this._notification = null;
+        });
+
+        // Now that we first detected this connection, mark it as such
+        let userSetting = this._ensureUserSetting(this._activeConnection);
+        userSetting.set_data(NM_SETTING_IS_NEW_CONNECTION, '1');
```

Comment:
As mentioned in Slack, I wonder if storing something more useful than just a boolean makes sense. Once added this `is-new-connection` data, we'll have to maintain it forever, so I wonder if perhaps storing something like a timestamp instead (and renaming to something like `last-notification-time`, for instance) could be equally helpful for this purpose, while perhaps being useful for something else in the future.